src/main/java/com/puppycrawl/tools/checkstyle/SuppressionsStringPrinter.java

```diff
@@ -0,0 +1,132 @@
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator;
+
+/**
+ * Class for constructing xpath queries to suppress nodes
+ * with specified line and column number.
+ * @author Timur Tibeyev.
+ */
+public final class SuppressionsStringPrinter {
+
+    /** Line and column number config value pattern. */
+    private static final Pattern VALID_SUPPRESSION_LINE_COLUMN_NUMBER_REGEX =
+            Pattern.compile("^[0-9]+:[0-9]+$");
+
+    /** Colon literal. */
+    private static final String COLON_LITERAL = ":";
+
+    /** Prevent instances. */
+    private SuppressionsStringPrinter() {
+        // no code
+    }
+
+    /**
+     * Prints generated suppressions.
+     * @param file the file to process.
+     * @param suppressionLineColumnNumber line and column number of the suppression
+     * @param tabWidth length of the tab character
+     * @return generated suppressions.
+     * @throws IOException if the file could not be read.
+     * @throws CheckstyleException if the file is not a Java source.
+     */
+    public static String printSuppressions(File file, String suppressionLineColumnNumber,
+                                           int tabWidth) throws IOException, CheckstyleException {
+        if (VALID_SUPPRESSION_LINE_COLUMN_NUMBER_REGEX.matcher(suppressionLineColumnNumber)
+                .matches()) {
+            final FileText fileText = new FileText(file.getAbsoluteFile(),
+                    System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
+            final DetailAST detailAST = parseFileText(fileText);
+            final int lineNumber = extractLineNumber(suppressionLineColumnNumber);
+            final int columnNumber = extractColumnNumber(suppressionLineColumnNumber);
```

Comment:
Why are we extracting the numbers manually? We did a regular expression match. Why don't we use that to our advantage to get the numbers for us.

Reply:
good catch.
In general parsing by simple substring is better then regexp usage. But we already run regexp .
So please update regexp to to have groups and keep values after "maches" executed.

Reply:
Done